respx/models.py

```diff
@@ -338,7 +338,7 @@ def _call_side_effect(
 
         try:
             # Call side effect
-            result: RouteResultTypes = effect(request, **kwargs)
+            result: RouteResultTypes = effect(request, is_async, **kwargs)
```

Comment:
To be fully backwards compatible, and not force users to change their existing side effects, I think we could be graceful here and only pass `is_async` among *kwargs* if the implemented side effect specs it, i.e. like the `route` kwarg is handled above.

e.g.
```py
if "is_async" in argspec.args:
            kwargs["is_async"] = is_async
```

.. maybe handle both spec'ed as arg or kwarg 🤔

Reply:
sounds good, done.